pathfs.go

```diff
@@ -220,6 +221,8 @@ func (p *PathFS) WriteFile(filename string, data []byte, perm os.FileMode) error
 // for the current path mainly because on Windows paths without a volume
 // specifier are never absolute, meaning that this check will always fail.
 func (p *PathFS) join(op string, name string) (string, error) {
+	// remove any volume names before appending the prefix
+	name = strings.Replace(name, filepath.VolumeName(name), "", 1)
```

Comment:
You can use string slicing to remove the prefix, as the length of the volume name is known:

```go
    if volumeName := filepath.VolumeName(name); volumeName != "" {
        name = name[len(volumeName):]
    }
```

The function's comment also needs updating to describe the new behavior.

Reply:
*cough* this can be code-golfed to:

```go
    // Strip any volume name.
    name = name[len(filepath.VolumeName(name)):]
```

but the suggestion in the previous comment is easier to read and probably compiles to the same code.

Reply:
I like the more explicit version too, so that's what I went with.